plugin.py

```diff
@@ -1,105 +1,73 @@
-
-        runCommand(
-            onCommandDone,
-            ["npm", "install", "--verbose", "--prefix", package_path, package_path]
-        )
-
-
-def onCommandDone():
-    logAndShowMessage('LSP-elm: Server installed.')
-
-
-def runCommand(onExit, popenArgs):
-    """
-    Runs the given args in a subprocess.Popen, and then calls the function
-    onExit when the subprocess completes.
-    onExit is a callable object, and popenArgs is a list/tuple of args that
-    would give to subprocess.Popen.
-    """
-    def runInThread(onExit, popenArgs):
-        try:
-            if sublime.platform() == 'windows':
-                subprocess.check_call(popenArgs, shell=True)
-            else:
-                subprocess.check_call(popenArgs)
-            onExit()
-        except subprocess.CalledProcessError as error:
-            logAndShowMessage('LSP-elm: Error while installing the server.', error)
-        return
-    thread = threading.Thread(target=runInThread, args=(onExit, popenArgs))
-    thread.start()
-    # returns immediately after the thread starts
-    return thread
+    server.setup()
 
 
-def is_node_installed():
-    return shutil.which('node') is not None
+def plugin_unloaded():
+    server.cleanup()
 
 
-def logAndShowMessage(msg, additional_logs=None):
-    print(msg, '\n', additional_logs) if additional_logs else print(msg)
-    sublime.active_window().status_message(msg)
+def is_node_installed():
+    return shutil.which('node') is not None
 
 
 class LspElmPlugin(LanguageHandler):
     @property
     def name(self) -> str:
-        return 'lsp-elm'
+        return PACKAGE_NAME.lower()
 
     @property
     def config(self) -> ClientConfig:
-        settings = sublime.load_settings("LSP-elm.sublime-settings")
-        client_configuration = settings.get('client')
+        # Calling setup() also here as this might run before `plugin_loaded`.
+        # Will be a no-op if already ran.
+        # See https://github.com/sublimelsp/LSP/issues/899
+        server.setup()
+
+        settings = sublime.load_settings(SETTINGS_FILENAME)
 
-        default_configuration = {
+        configuration = {
+            "enabled": settings.get("enabled", True),
             "command": [
                 'node',
-                server_path,
+                server.binary_path,
                 '--stdio'
             ],
-            "languages": [
+            "languages": settings.get("languages", [
                 {
                     "languageId": "elm",
                     "scopes": ["source.elm"],
                     "syntaxes": ["Packages/Elm Syntax Highlighting/src/elm.sublime-syntax"]
                 }
-            ],
-            "initializationOptions": {
+            ]),
+            "initializationOptions": settings.get("initializationOptions", {
                 "elmAnalyseTrigger": "change"
-            }
+            }),
+            "settings": settings.get("settings", {})
```

Comment:
We can remove `languages`, `initializationOptions` and `settings`, as we did in other packages.
If user doesn't override them, they will be taken from default settings. If he does, then user's will be used.